### PR TITLE
make the drainer name configurable

### PR DESCRIPTION
**Pull request by gregwebs**
This is used for the statefulset/pod names.

The release name is already unique, so I would actually suggest just using that
without the cluster name. However, that is a backwards incompatible
change that I hesitate to make.

Consider this more of an issue than a pull request. I haven't tested this yet.

Code changes

- chart

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Allow drainerName to be configured in the tidb-drainer chart.
```

### CODE REVIEW

**Comment by gregwebs** [comment]
Right now this is backwards compatible so I don't know what you are suggesting.

File: charts/tidb-drainer/values.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@
 # timezone is the default system timzone
 timezone: UTC
 
+# Change the name of the statefulset and pod
+# The default is clusterName-ReleaseName-drainer
```
Comment (aylei) [inline charts/tidb-drainer/values.yaml]: Can we warn user that changing name on existing will cause data loss?

Comment (cofyc) [reply charts/tidb-drainer/values.yaml]: cc @gregwebs add a notice then we merge this?

Comment (gregwebs) [reply charts/tidb-drainer/values.yaml]: okay, warning added

**Pull request merged**